libfwupdplugin/fu-progress.c

```diff
@@ -334,6 +330,43 @@ fu_progress_get_percentage(FuProgress *self)
 	return priv->percentage;
 }
 
+static void
+fu_progress_set_parent(FuProgress *self, FuProgress *parent)
+{
+	FuProgressPrivate *priv = GET_PRIVATE(self);
+	g_return_if_fail(FU_IS_PROGRESS(self));
```

Comment:
I think you want to fail if parent is null too, no?

Reply:
Agree, added.